extensions/vscode/src/api/resources/Configurations.ts

```diff
@@ -20,7 +20,7 @@ export class Configurations {
   // 200 - success
   // 404 - not found
   // 500 - internal server error
-  get(configName: string, params?: { dir?: string }) {
+  get(configName: string, params: { dir: string }) {
```

Comment:
I understand this is to avoid us from forgetting the `dir` param when we should include it, but I do think we should discuss the value we are losing here. With this change the `api` module diverges from representing the API.

Personally I would prefer the module be a 1-1 with the actual API functionality. Mainly because I believe it causes confusion for anyone who isn't intimately familiar with the patterns of this code base why a URL param (which is normally optional) is not optional here.

Reply:
Let's discuss.